tests/platform_tests/fwutil/conftest.py

```diff
@@ -0,0 +1,107 @@
+        f.extractall("./firmware/")
+        with open('./firmware/firmware.json', 'r') as fw:
+            fw_data = json.load(fw)
+            yield fw_data
+    subprocess.call("rm -rf firmware", shell=True)
+
+@pytest.fixture(scope='function')
+def random_component(duthost, fw_pkg):
+    chass = show_firmware(duthost)["chassis"].keys()[0]
+    components = fw_pkg["chassis"].get(chass, {}).get("component", []).keys()
+
+    if len(components) == 0:
+        pytest.skip("No suitable components found in config file for platform {}.".format(duthost.facts['platform']))
+
+    return components[randrange(len(components))] 
+
+@pytest.fixture(scope='function')
+def host_firmware(localhost, duthost):
+    logger.info("Starting local python server to test URL firmware update....")
+    comm = "python3 -m http.server --directory {}".format(os.path.join(DEVICES_PATH, 
+        duthost.facts['platform']))
+    task, res = duthost.command(comm, module_ignore_errors=True, module_async=True)
+    yield "http://localhost:8000/"
+    task.terminate()
+
+@pytest.fixture(scope='function')
+def next_image(duthost, fw_pkg):
+
+    # Install next version of sonic
+    current = duthost.shell('sonic_installer list | grep Current | cut -f2 -d " "')['stdout']
+
+    image = fw_pkg.get("images", {}).keys()
+    target = None
+
+    for i in image:
+        if "SONiC-OS-{}".format(i) != current:
+            target = i
+
+    if target is None:
+        pytest.skip("No suitable image definitions found in config")
+
+    logger.info("Installing new image {}".format(target))
+    duthost.copy(src=os.path.join("firmware", fw_pkg["images"][target]), dest=DUT_HOME)
+    remote_path = os.path.join(DUT_HOME, os.path.basename(fw_pkg["images"][target]))
+    duthost.command("sonic_installer install -y {}".format(remote_path), module_ignore_errors=True)
+
+    # Mount newly installed image
+    fs_path = FS_PATH_TEMPLATE.format(target)
+    fs_mountpoint = FS_MOUNTPOINT_TEMPLATE.format(target)
+    fs_rw = FS_RW_TEMPLATE.format(target)
+    fs_work = FS_WORK_TEMPLATE.format(target)
+    overlay_mountpoint = OVERLAY_MOUNTPOINT_TEMPLATE.format(target)
+
+    logger.info("Attempting to stage test firware onto newly-installed image.")
+    try:
+        wait_until(10, 1, check_path_exists, fs_rw)
+
+        duthost.command("mkdir -p {}".format(fs_mountpoint))
+        cmd = "mount -t squashfs {} {}".format(fs_path, fs_mountpoint)
+        duthost.command(cmd)
+
+        duthost.command("mkdir -p {}".format(overlay_mountpoint))
+        cmd = "mount -n -r -t overlay -o lowerdir={},upperdir={},workdir={},rw overlay {}".format(
+            fs_mountpoint,
+            fs_rw,
+            fs_work,
+            overlay_mountpoint
+        )
+        duthost.command(cmd)
+    except Exception as e:
+        pytest.fail("Failed to setup next-image.")
+        duthost.command("sonic_installer set-default {}".format(current))
+
+    yield overlay_mountpoint
```

Comment:
@alexrallen we might need a teardown here when config file contains only single component which supports on-the-fly FW upgrade. In that case, the switch will be left with next image mounted.

Reply:
This is correct, just saw this issue in regression actually. Fixed.